airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -218,9 +214,61 @@ private static Container getMain(final String image,
     return containerBuilder.build();
   }
 
-  private static void copyFilesToKubeConfigVolume(final KubernetesClient client,
-                                                  final Pod podDefinition,
-                                                  final Map<String, String> files) {
+  public static List<ContainerPort> createContainerPortList(final Map<Integer, Integer> internalToExternalPorts) {
+    return internalToExternalPorts.keySet().stream()
+        .map(integer -> new ContainerPortBuilder()
+            .withContainerPort(integer)
+            .build())
+        .collect(Collectors.toList());
+  }
+
+  public static void copyFilesToKubeConfigVolumeMain(final Pod podDefinition, final Map<String, String> files) {
+    final List<Map.Entry<String, String>> fileEntries = new ArrayList<>(files.entrySet());
+
+    // copy this file last to indicate that the copy has completed
+    fileEntries.add(new AbstractMap.SimpleEntry<>(SUCCESS_FILE_NAME, ""));
+
+    Path tmpFile = null;
+    Process proc = null;
+    for (final Map.Entry<String, String> file : fileEntries) {
+      try {
+        tmpFile = Path.of(IOs.writeFileToRandomTmpDir(file.getKey(), file.getValue()));
+
+        LOGGER.info("Uploading file: " + file.getKey());
+        final var containerPath = Path.of(CONFIG_DIR + "/" + file.getKey());
+
+        // using kubectl cp directly here, because both fabric and the official kube client APIs have
+        // several issues with copying files. See https://github.com/airbytehq/airbyte/issues/8643 for
+        // details.
+        final String command = String.format("kubectl cp %s %s/%s:%s -c %s", tmpFile, podDefinition.getMetadata().getNamespace(),
+            podDefinition.getMetadata().getName(), containerPath, "main");
+        LOGGER.info(command);
+
+        proc = Runtime.getRuntime().exec(command);
+        LOGGER.info("Waiting for kubectl cp to complete");
+        final int exitCode = proc.waitFor();
+
+        if (exitCode != 0) {
+          throw new IOException("kubectl cp failed with exit code " + exitCode);
+        }
+
+        LOGGER.info("kubectl cp complete, closing process");
+      } catch (final IOException | InterruptedException e) {
+        throw new RuntimeException(e);
+      } finally {
+        if (tmpFile != null) {
+          tmpFile.toFile().delete();
```

Comment:
Can this line throw an exception if the file doesn't exist and thus prevent the proc to be destroyed?